pkg/cloud/api/fill.go

```diff
@@ -179,7 +179,7 @@ func fillNullAndForceSend(traits *FieldTraits, v reflect.Value) error {
 			fType := traits.fieldType(p.Field(ft.Name))
 			fv := v.Field(i)
 
-			if fType == FieldTypeOrdinary {
+			if fType == FieldTypeMandatory {
```

Comment:
Fill's contract is to fill __all__ field values, not just Mandatory ones. If this is not clear, please update the Docstring.

Reply:
I changed the name from Mandatory to NonZeroValue so maybe this way it will be clear.
In this check we are adding fields to NullFields or ForceSendFields if they are nil. We don't have to do this for fields that are allowed to be zero